lib/pacemaker/pcmk_sched_native.c

```diff
@@ -1338,8 +1337,16 @@ native_create_actions(pe_resource_t * rsc, pe_working_set_t * data_set)
             crm_notice("See https://wiki.clusterlabs.org/wiki/FAQ#Resource_is_Too_Active for more information");
         }
 
-        if (rsc->recovery_type == recovery_stop_start) {
-            need_stop = TRUE;
+        switch (rsc->recovery_type) {
+            case recovery_stop_start:
+                need_stop = TRUE;
```

Comment:
need_stop could similarly just be a bool, and in fact it's passed to schedule_restart_actions which takes a bool as an argument.